### PR TITLE
Add action for installing and testing pyopenms nightlies on different platforms

### PR DESCRIPTION
**Pull request by matteopilz**
## Description
I couldn't test the automatic trigger. Additionally, we could create a test Python script file instead of including the code in the .yml.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: .github/workflows/test_pyopenms.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - completed
+
+jobs:
+  test:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (jpfeuffer) [inline .github/workflows/test_pyopenms.yml:17]: Cool. But this will not work, since our wheels are not standardized by versions.
See #6704

Comment (jpfeuffer) [reply .github/workflows/test_pyopenms.yml:17]: If you check the pyopenms-wheels job, you will see that linux builds different versions for example.

Comment (jpfeuffer) [reply .github/workflows/test_pyopenms.yml:17]: And it is not very maintainable to have multiple places in which to change versions.

Comment (matteopilz) [reply .github/workflows/test_pyopenms.yml:17]: So what should we do, test one Python version, e.g. the latest we build for?

Comment (jpfeuffer) [reply .github/workflows/test_pyopenms.yml:17]: No. Introduce a file with list of python versions to build. That is used by all CI. See #6704

Comment (jpfeuffer) [reply .github/workflows/test_pyopenms.yml:17]: See e.g. https://www.cynkra.com/blog/2020-12-23-dynamic-gha/ how to create a GHA matrix dynamically.

Comment (jpfeuffer) [reply .github/workflows/test_pyopenms.yml:17]: The pyopenms-wheels scripts can just read it in shell

File: .github/workflows/test_pyopenms.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - completed
+
+jobs:
+  read-python-versions:
+    steps:
+      - name: Get values from python_versions.json
+        id: python_versions
+        uses: RadovanPelka/github-action-json@main
+        with:
+          path: "pyton_versions.json"
```
Comment (jpfeuffer) [inline .github/workflows/test_pyopenms.yml:17]:
```suggestion
          path: "python_versions.json"
```

**Comment by matteopilz** [comment]
I'm trying to figure out a way to solve this. It seems the matrix needs to be defined at job level to be used as strategy.

**Comment by jpfeuffer** [comment]
I posted a link earlier. That did not work?

**Comment by jpfeuffer** [comment]
It shows that you need two jobs inside your workflow. One that creates the matrix values. And the second one that uses it.

**Comment by matteopilz** [comment]
Yes, the catch is, that the first job already needs the defined `runs-on`. So I'm not sure if it will work, but I can try it out.

**Comment by matteopilz** [comment]
Looks like it works, I wanted to prevent setting up another environment beforehand, but I guess it doesn't matter.
I will add the nightly build trigger again.

**Comment by jpfeuffer** [comment]
Spawning Ubuntu latest is super fast if you don't install any stuff. Should be no problem.

**Review by timosachsenberg — APPROVED** [review]
Looks good to me! Thanks 🙏

**Comment by timosachsenberg** [comment]
after the merge it can now be triggered
image

**Comment by timosachsenberg** [comment]
@KyowonJeong a good example of a not too complicated github action

**Comment by matteopilz** [comment]
I also wanted to put in a trigger for push on the nightly, py_nightly_test branches, after the  pyopenms-wheels action is done. Should I just leave this out now?

**Comment by jpfeuffer** [comment]
No, would be good. You had it in there at some point, no?

**Comment by jpfeuffer** [comment]
Maybe you can make the pyopenms-wheels branch trigger this workflow.

**Comment by jpfeuffer** [comment]
pyopenms-wheels should also use the python-versions file now

**Comment by matteopilz** [comment]
Yes, I had it in there and removed it completely to make testing easier. I will add it and open another PR. I can also setup pyopenms-wheels to read the python-versions.

**Pull request merged**